src/index.rs

```diff
@@ -231,8 +235,63 @@ impl IndexTable {
 		Ok(try_io!(Ok(&map[offset..offset + CHUNK_LEN])))
 	}
 
-	#[inline(never)]
 	fn find_entry(&self, key_prefix: u64, sub_index: usize, chunk: &[u8]) -> (Entry, usize) {
+		#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+		if is_x86_feature_detected!("sse2") {
+			return self.find_entry_sse2(key_prefix, sub_index, chunk)
+		}
+		self.find_entry_regular(key_prefix, sub_index, chunk)
+	}
+
+	#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+	fn find_entry_sse2(&self, key_prefix: u64, sub_index: usize, chunk: &[u8]) -> (Entry, usize) {
+		debug_assert!(
```

Comment:
`sub_index` is the starting index for the search, rather than the number of entities. There may be multiple entries mathing the partial key in a single chunk. `find_entry` is called in a loop and all matching entries are further validated for the full key that's stored in the value table. `sub_index` is set to the entry index of the previous match, so that the search continues if the previous match did not pass that validation. Therefore it can be arbitatrary. We can just round it here to the prevous multiple of four. E.g. if `sub_index` is 7, set it to 4 and skip the first 3 result bits. 

I think we can also use aligned loads in this case. Though this will probably require changing the type from `IndexChunk` to a struct with alignment attribute here:
https://github.com/paritytech/parity-db/blob/master/src/log.rs#L444

Reply:
Done.